beater/beater.go

```diff
@@ -3,37 +3,37 @@ package beater
 import (
 	"fmt"
 
-	"github.com/elastic/apm-server/config"
 	"github.com/elastic/apm-server/server"
 	"github.com/elastic/beats/libbeat/beat"
 	"github.com/elastic/beats/libbeat/common"
 	"github.com/elastic/beats/libbeat/logp"
 )
 
-type ApmServer struct {
+type ApmBeater struct {
 	done   chan struct{}
-	server *server.Server
-	config config.Config
+	server *server.ApmServer
+	config BeaterConfig
 	client beat.Client
 }
 
 // Creates beater
-func New(_ *beat.Beat, cfg *common.Config) (beat.Beater, error) {
-	config := config.DefaultConfig
-	if err := cfg.Unpack(&config); err != nil {
+func New(_ *beat.Beat, ucfg *common.Config) (beat.Beater, error) {
```

Comment:
I don't think there is a risk (also `http` is a package name and code was working with var named name `http`)
`cfg` / `config` / `conf` is too generic,  this way I know what type I have. `cfg` and `config` was a big part of my confusion before.